pkg/engine/templates_generated.go

```diff
@@ -35797,13 +35797,15 @@ var _k8sCloudInitArtifactsMountetcdSh = []byte(`#!/bin/bash
 # Mounting is done here instead of etcd because of bug https://bugs.launchpad.net/cloud-init/+bug/1692093
 # Once the bug is fixed, replace the below with the cloud init changes replaced in https://github.com/Azure/aks-engine/pull/661.
 set -x
+ETCDDISK=""
 PARTITION=""
 for DISK in $(cat /proc/partitions | grep -o sd[a-z] | uniq); do
   if ! cat /proc/partitions | grep "$DISK"1; then
     if [[ -n $PARTITION ]]; then
       exit 1
     fi
-    PARTITION=/dev/${DISK}1
+    ETCDDISK=/dev/${DISK}
+    PARTITION=${ETCDDISK}1
   fi;
 done
 MOUNTPOINT=/var/lib/etcddisk
```

Comment:
You should check that ETCDDISK is set by the time you get out of the loop.  Just in case there was no disk...
(That would be bad, but it would be worse to use "" as the disk...)